zrpc/resolver/internal/kubebuilder.go

```diff
@@ -61,27 +69,39 @@ func (b *kubeBuilder) Build(target resolver.Target, cc resolver.ClientConn,
 		}); err != nil {
 			logx.Error(err)
 		}
-	})
-	inf := informers.NewSharedInformerFactoryWithOptions(cs, resyncInterval,
-		informers.WithNamespace(svc.Namespace),
-		informers.WithTweakListOptions(func(options *v1.ListOptions) {
-			options.FieldSelector = nameSelector + svc.Name
-		}))
-	in := inf.Core().V1().Endpoints()
-	_, err = in.Informer().AddEventHandler(handler)
-	if err != nil {
-		return nil, err
 	}
 
-	threading.GoSafe(func() {
-		inf.Start(proc.Done())
-	})
-	endpoints, err := cs.CoreV1().Endpoints(svc.Namespace).Get(context.Background(), svc.Name, v1.GetOptions{})
-	if err != nil {
-		return nil, err
-	}
+	if !ok {
+		handler = kube.NewEventHandler(updates)
 
-	handler.Update(endpoints)
+		inf := informers.NewSharedInformerFactoryWithOptions(cs, resyncInterval,
+			informers.WithNamespace(svc.Namespace),
+			informers.WithTweakListOptions(func(options *v1.ListOptions) {
+				options.FieldSelector = nameSelector + svc.Name
+			}))
+		in := inf.Core().V1().Endpoints()
+		_, err = in.Informer().AddEventHandler(handler)
+		if err != nil {
+			return nil, err
+		}
+
+		threading.GoSafe(func() {
+			inf.Start(proc.Done())
+		})
+
+		endpoints, err := cs.CoreV1().Endpoints(svc.Namespace).Get(context.Background(), svc.Name, v1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		handler.Update(endpoints)
+
+		b.lock.Lock()
+		b.handlers[target] = handler
+		b.lock.Unlock()
+	} else {
+		updates(handler.Targets())
```

Comment:
If handler already exists, why do we use the previous targets in handler?

Reply:
> If handler already exists, why do we use the previous targets in handler?

I thought `cc resolver.ClientConn` was brand new for new `grpc.ClientConn`, but it looks like it was not. So if event handler of current target existed, just ignore this `Build` call.